android/src/main/java/com/vydia/RNUploader/ChunkUtils.kt

```diff
@@ -0,0 +1,56 @@
+package com.vydia.RNUploader
+
+import com.facebook.react.bridge.ReadableArray
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+import java.io.FileInputStream
+import java.io.FileOutputStream
+
+
+data class Chunk(val position: Long, val size: Long, val path: String) {
```

Comment:
this whole file was moved from `Uitls.kt`. Only the `chunkFile` method was changed to replace unsafe use of memory mapping with buffer and streams, which still produce the same fast performance.